yt_dlp/utils.py

```diff
@@ -3275,6 +3275,8 @@ def strip_jsonp(code):
 
 def js_to_json(code, vars={}, *, strict=False):
     # vars is a dict of var, val pairs to substitute
+    STRING_QUOTES = '\'\"'
+    STRING_RE = '|'.join(rf'''{q}(?:[^{q}\\]*(?:\\.))*[^{q}\\]*{q}''' for q in STRING_QUOTES)
```

Comment:
```suggestion
    STRING_RE = '|'.join(rf'{q}(?:[^{q}\\]*(?:\\.))*[^{q}\\]*{q}' for q in STRING_QUOTES)
```
OR
untested, but this should work better
```suggestion
    STRING_RE = '|'.join(rf'{q}(?:(?<!\\)(?:\\\\)*\\{q}|[^{q}])*{q}' for q in STRING_QUOTES)
```

Reply:
Actually, whats speaking against `rf'{q}(?:\\.|[^\\{q}])*{q}'`?